MapleServer2/Types/StatDistribution.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MapleServer2.Types
+{
+    public class StatDistribution
+    {
+        public int TotalStatPoints { get; private set; }
+        // TODO: implement Dictionary to keep track of points earned from quest, trophy, exploration, prestige
+        // naming convention: PointsFromQuest, PointsFromTrophy, PointsFromExploration, PointsFromPrestige
+
+        public Dictionary<byte, int> AllocatedStats { get; private set; }
+        // key = index representing the stat type (ie. a value of 00 corresponds to Str)
+        // value = number of points allocated to the stat
+
+        public StatDistribution()
+        {
+            // hardcode the amount of stat points the character starts with temporarily
+            this.TotalStatPoints = 18;
+            this.AllocatedStats = new Dictionary<byte, int>();
+        }
+
+        public StatDistribution(Dictionary<byte, int> AllocatedStats)
+        {
+            this.AllocatedStats = AllocatedStats;
+        }
+
+        public void AddPoint(byte StatType)
+        {
+            int statCount;
+            if (AllocatedStats.TryGetValue(StatType, out statCount))
+            {
+                AllocatedStats[StatType] = statCount + 1;
```

Comment:
Would this work by just doing `AllocatedStats[StatType] += 1`? And if so, we can then remove the need for `statCount` altogether and just check for the existence of the stat

Reply:
Changing my implementation as suggested - initially, I thought TryGetValue would be more efficient because it performs better when the key is frequently not found (as would be the case when a user starts with or resets to an empty stat distribution).

Since there are only 6 stats and players usually only allocate to 1 or 2 stats, ContainsKey would perform better and the implementation should be more readable to non C# coders.